apps/condo/domains/meter/schema/MeterReading.js

```diff
@@ -123,6 +142,37 @@ const MeterReading = new GQLListSchema('MeterReading', {
 
             return resolvedData
         },
+        validateInput: async ({ context, resolvedData, existingItem }) => {
+            const newItem = { ...existingItem, ...resolvedData }
+            const locale = extractReqLocale(context.req) || conf.DEFAULT_LOCALE
+
+            const meterId = get(newItem, 'meter')
+
+            const meter = await getById('Meter', meterId)
+
+            const emptyFieldsNames = []
+            for (let i = 1; i <= meter.numberOfTariffs; i++) {
+                if (isNil(get(newItem, `value${i}`))) {
+                    emptyFieldsNames.push(`value${i}`)
+                }
+            }
+            if (emptyFieldsNames.length > 0) {
+                const localizedFieldsNames = emptyFieldsNames.map((fieldName) => i18n(`meter.import.column.${fieldName}`, { locale }))
+                throw new GQLError(ERRORS.METER_READING_FEW_VALUES(meter.number, meter.numberOfTariffs, localizedFieldsNames.join(', ')), context)
+            }
+
+            const extraFieldsNames = []
+            for (let i = meter.numberOfTariffs + 1; i <= VALUES_COUNT; i++) {
+                const value = get(newItem, `value${i}`)
+                if (!isNil(value) && !isEmpty(value)) {
+                    extraFieldsNames.push(`value${i}`)
+                }
+            }
+            if (extraFieldsNames.length > 0) {
+                const localizedFieldsNames = extraFieldsNames.map((fieldName) => i18n(`meter.import.column.${fieldName}`, { locale }))
```

Comment:
import?

Reply:
Yes. To make error understandable during registry uploading.

Reply:
A bit strange translation key. Not obvious, that if I'll change it - Readings will be changed too

Reply:
This error is useful for excel import. Frontend has it's own validation.